### PR TITLE
feat(tianmu): support alter table xxx modify xxx type NULL/NOT NULL

### LINKED ISSUE #1671: backport: support alter table xxx modify xxx type NULL/NOT NULL
**Is your feature request related to a problem? Please describe.**

previous working issue: #860
commit id: d0c2e01b6e5c0ccca6cc0f6aa1a5568d7b08c3a6

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by chenshengjiang**
## Summary about this PR

support field properties from NULL to NOT NULL
support field properties from NOT NULL to NULL
Issue Number: close #1671

## Tests Check List

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [x] New Feature
- [ ] Bug Fix
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

File: storage/tianmu/handler/ha_tianmu.cpp [hunk 1]
```diff
@@ -614,15 +614,15 @@ int ha_tianmu::rename_table(const char *from, const char *to, const dd::Table *f
                             dd::Table *to_table_def) {  // stonedb8 TODO
   try {
     ha_rcengine_->RenameTable(current_txn_, from, to, from_table_def, to_table_def, ha_thd());
-    return 0;
+    DBUG_RETURN(0);
```
Comment (hustjieke) [inline storage/tianmu/handler/ha_tianmu.cpp]: return value is `bool`, fix 
`0`--->`false`,
`1`--->`true`

File: storage/tianmu/handler/ha_tianmu.cpp [hunk 2]
```diff
@@ -614,15 +614,15 @@ int ha_tianmu::rename_table(const char *from, const char *to, const dd::Table *f
...
   } catch (std::exception &e) {
     my_message(static_cast<int>(common::ErrorCode::UNKNOWN_ERROR), e.what(), MYF(0));
     TIANMU_LOG(LogCtl_Level::ERROR, "An exception is caught: %s", e.what());
   } catch (...) {
     my_message(static_cast<int>(common::ErrorCode::UNKNOWN_ERROR), "An unknown system exception error caught.", MYF(0));
     TIANMU_LOG(LogCtl_Level::ERROR, "An unknown system exception error caught.");
   }
-  return 1;
+  DBUG_RETURN(1);
```
Comment (hustjieke) [inline storage/tianmu/handler/ha_tianmu.cpp]: return value is `bool`, fix 
`0`--->`false`,
`1`--->`true`

File: storage/tianmu/handler/ha_tianmu.cpp [hunk 3]
```diff
@@ -769,8 +769,9 @@ int ha_tianmu::open(const char *name, [[maybe_unused]] int mode, [[maybe_unused]
 }
 
 int ha_tianmu::free_share() {
+  DBUG_ENTER(__PRETTY_FUNCTION__);
   share_.reset();
-  return 0;
+  DBUG_RETURN(0);
```
Comment (hustjieke) [inline storage/tianmu/handler/ha_tianmu.cpp:777]: return value is `bool`, fix 
`0`--->`false`,
`1`--->`true`

Comment (chenshengjiang) [reply storage/tianmu/handler/ha_tianmu.cpp:777]: here return value is int

**Review by hustjieke — APPROVED** [review]
LGTM

**Review by Double0101 — APPROVED** [review]
LGTM

**Pull request merged**